src/js/themes/base.d.ts

```diff
@@ -1257,7 +1258,7 @@ export interface ThemeType {
       size?: string;
     };
     icons?: {
-      circle?: string;
+      circle?: any;
```

Comment:
Eventually it likely makes sense to provide stronger typing for this and things like `checkBox.icons.checked`...

Appears as though there are a few variants inside of the `ThemeType` currently e.g.

```typescript
JSX.Element | string
```
[here](https://github.com/grommet/grommet/blob/master/src/js/themes/base.d.ts#L242)

```typescript
React.ReactNode | Icon
```
[here](https://github.com/grommet/grommet/blob/master/src/js/themes/base.d.ts#L1011-L1014)

```typescript
ReactComponentElement<any>;
```
[here](https://github.com/grommet/grommet/blob/master/src/js/themes/base.d.ts#L1363)


Considering something like `React.ReactNode | Icon` or `React.ReactNode | Icon | string` would allow for both flexibility in implementation as well as more informative type hinting for developers.

Reply:
I think we can go ahead and use `React.ReactNode | Icon` as our type here

Reply:
Made these updates @jcfilben, thank you!

Reply:
@jcfilben what is the consideration for dropping the type of `string`? wouldn't that be important for backward compatibility?

Reply:
What `string` value could be passed here? I was thinking we could remove the `string` type here because I wasn't sure that there would be any strings that could be passed here.

Reply:
When playing around with this previously during my debugging I noticed the only strings that appear to work are names of HTML tags e.g. `span`, `div` etc..

However when these are rendered in the DOM they are empty tags and are invisible to the user.  We certainly could keep `string` here to maintain backward compatibility.   That said, anyone who is currently using this is likely writing custom CSS styles to manually insert an icon via `background-image` or something similar.

This in combination that the CSS classes generated by styled components are often unpredictable and done during build/runtime I have a hard time believing anyone is currently using this approach 💭

Reply:
Thanks for the description, this is helpful. I think since someone could pass a string here even though it may be unlikely, we can keep `string` for backwards compatibility. So `React.ReactNode | Icon | string`